pkg/operator/staticpod/installerpod/cmd.go

```diff
@@ -136,144 +150,164 @@ func (o *InstallOptions) prefixFor(name string) string {
 	return name[0 : len(name)-len(fmt.Sprintf("-%s", o.Revision))]
 }
 
-func (o *InstallOptions) copyContent() error {
-	// gather all secrets
+func (o *InstallOptions) copyContent(ctx context.Context) error {
+	secretPrefixes := sets.NewString(o.SecretNamePrefixes...)
+	optionalSecretPrefixes := sets.NewString(o.OptionalConfigMapNamePrefixes...)
+	configPrefixes := sets.NewString(o.ConfigMapNamePrefixes...)
+	optionalConfigPrefixes := sets.NewString(o.OptionalConfigMapNamePrefixes...)
+
+	// Gather secrets. If we get API server error, retry getting until we hit the timeout.
+	// Retrying will prevent temporary API server blips or networking issues.
+	// We return when all "required" secrets are gathered, optional secrets are not checked.
 	secrets := []*corev1.Secret{}
-	for _, currPrefix := range o.SecretNamePrefixes {
-		glog.Infof("getting secrets/%s -n %s", o.nameFor(currPrefix), o.Namespace)
-		val, err := o.KubeClient.CoreV1().Secrets(o.Namespace).Get(o.nameFor(currPrefix), metav1.GetOptions{})
-		if err != nil {
-			return err
+	err := utilwait.PollImmediateUntil(200*time.Millisecond, func() (bool, error) {
+		for _, prefix := range append(secretPrefixes.List(), optionalSecretPrefixes.List()...) {
+			secret, err := o.KubeClient.CoreV1().Secrets(o.Namespace).Get(o.nameFor(prefix), metav1.GetOptions{})
+			if err != nil {
+				glog.Infof("Failed to get secret %s/%s: %v", o.Namespace, o.nameFor(prefix), err)
+				return false, nil
+			}
+			secretPrefixes.Delete(prefix)
+			secrets = append(secrets, secret)
 		}
-		secrets = append(secrets, val)
+
+		// Exit when all required secrets are gathered
+		return secretPrefixes.Len() == 0, nil
+	}, ctx.Done())
+	if err != nil {
+		return err
 	}
-	for _, currPrefix := range o.OptionalSecretNamePrefixes {
-		glog.Infof("getting optional secrets/%s -n %s", o.nameFor(currPrefix), o.Namespace)
-		val, err := o.KubeClient.CoreV1().Secrets(o.Namespace).Get(o.nameFor(currPrefix), metav1.GetOptions{})
-		if errors.IsNotFound(err) {
-			glog.Infof("missing optional secrets/%s -n %s", o.nameFor(currPrefix), o.Namespace)
-			continue
-		}
-		if err != nil {
-			return err
+
+	// Gather all config maps
+	configs := []*corev1.ConfigMap{}
+	err = utilwait.PollImmediateUntil(200*time.Millisecond, func() (bool, error) {
+		for _, prefix := range append(configPrefixes.List(), optionalConfigPrefixes.List()...) {
+			config, err := o.KubeClient.CoreV1().ConfigMaps(o.Namespace).Get(o.nameFor(prefix), metav1.GetOptions{})
+			if err != nil {
+				glog.Infof("Failed to get config map %s/%s: %v", o.Namespace, o.nameFor(prefix), err)
```

Comment:
now this retries OptionalConfigMapNamePrefixes, but probably should ignore NotFounds.